File: app-v3/src/main/java/com/khmelenko/lab/varis/auth/AuthActivityModule.java
package com.khmelenko.lab.varis.auth;

import android.app.Activity;

import dagger.Binds;
import dagger.Module;
import dagger.android.ActivityKey;
import dagger.android.AndroidInjector;
import dagger.multibindings.IntoMap;

/**
 * @author Dmytro Khmelenko (d.khmelenko@gmail.com)
 */
@Module(subcomponents = AuthActivitySubcomponent.class)
public abstract class AuthActivityModule {
    @Binds
    @IntoMap
    @ActivityKey(AuthActivity.class)
    abstract AndroidInjector.Factory<? extends Activity>
    bindAuthActivityInjectorFactory(AuthActivitySubcomponent.Builder builder);
}

File: app-v3/src/main/java/com/khmelenko/lab/varis/auth/AuthActivityModule.kt
package com.khmelenko.lab.varis.auth

import android.app.Activity

import dagger.Binds
import dagger.Module
import dagger.Subcomponent
import dagger.android.ActivityKey
import dagger.android.AndroidInjector
import dagger.multibindings.IntoMap

@Module(subcomponents = [(AuthActivitySubcomponent::class)])
abstract class AuthActivityModule {
    @Binds
    @IntoMap
    @ActivityKey(AuthActivity::class)
    internal abstract fun bindAuthActivityInjectorFactory(
            builder: AuthActivitySubcomponent.Builder): AndroidInjector.Factory<out Activity>
}

@Subcomponent
interface AuthActivitySubcomponent : AndroidInjector<AuthActivity> {
    @Subcomponent.Builder
    abstract class Builder : AndroidInjector.Builder<AuthActivity>()
}
